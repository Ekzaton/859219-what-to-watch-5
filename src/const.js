export const APIRoute = {
  FILMS: `/films/`,
  FAVORITE: `/favorite/`,
  PROMO: `/films/promo/`,
  COMMENTS: `/comments/`,
  LOGIN: `/login/`
};

export const AppRoute = {
  ROOT: `/`,
  SIGN_IN: `/login/`,
  MY_LIST: `/mylist/`,
  REVIEW: `/review/`,
  REVIEW_ID: `/films/:id/review/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id/`,
  FILMS: `/films/`,
  FILMS_ID: `/films/:id/`
};

export const Tab = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNRATED: `Unrated`
};

export const Review = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 400,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const DEFAULT_GENRE = `All genres`;

export const DEFAULT_MOVIES_COUNT = 8;
