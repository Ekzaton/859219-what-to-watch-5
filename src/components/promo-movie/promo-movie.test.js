import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Route, BrowserRouter} from "react-router-dom";

import {AuthorizationStatus} from "../../const";

import mockPromo from "../../mocks/promo";

import PromoMovie from "./promo-movie";

describe(`PromoMovie snapshot testing`, () => {
  it(`(withot Auth) renders store-connected component correctly`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_DATA: {
        promoMovie: mockPromo,
      },
      APP_USER: {
        status: AuthorizationStatus.NO_AUTH,
      },
    });

    const promoMovieComponent = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <PromoMovie/>
            </Route>
          </BrowserRouter>
        </Provider>
    );

    expect(promoMovieComponent.toJSON()).toMatchSnapshot();
  });

  it(`(with Auth) renders store-connected component correctly`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_DATA: {
        promoMovie: mockPromo,
      },
      APP_USER: {
        status: AuthorizationStatus.AUTH,
      },
    });

    const promoMovieComponent = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <PromoMovie/>
            </Route>
          </BrowserRouter>
        </Provider>
    );

    expect(promoMovieComponent.toJSON()).toMatchSnapshot();
  });
});
