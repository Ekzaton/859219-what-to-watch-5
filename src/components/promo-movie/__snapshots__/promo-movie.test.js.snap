// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoMovie snapshot testing renders component correctly 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Bronson"
      src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <div
        className="user-block__link"
      >
        <a
          className="user-block__link"
          href="/sign-in/"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="Bronson poster"
          height="327"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Bronson
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Action
          </span>
          <span
            className="movie-card__year"
          >
            2008
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
